subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -611,10 +640,11 @@ static uint8_t service_discover_func(struct bt_conn *conn,
 		if (err != 0) {
 			LOG_DBG("Discover failed (err %d)", err);
 			broadcast_assistant.busy = false;
-
-			if (broadcast_assistant_cbs != NULL &&
-			    broadcast_assistant_cbs->discover != NULL) {
-				broadcast_assistant_cbs->discover(conn, err, 0);
+			SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&broadcast_assistant_cbs,
+							  listener, next, _node) {
+				if (listener->discover) {
+					listener->discover(conn, err, 0);
+				}
```

Comment:
Suggest to create helper functions for calling the callback, as we've done for e.g. VCP and MICP: 
```c
static void vcp_vol_ctlr_flags_changed(struct bt_vcp_vol_ctlr *vol_ctlr, int err)
{
	struct bt_vcp_vol_ctlr_cb *listener, *next;

	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&vcp_vol_ctlr_cbs, listener, next, _node) {
		if (listener->flags) {
			listener->flags(vol_ctlr, err, err == 0 ? vol_ctlr->flags : 0);
		}
	}
}
```

This applies for all callbacks

Reply:
I didn't update the ones in the bap_broadcast_assistant_write_cp_cb, these are called only once. This is similar to what was done for VCP